main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	kustomizeapi "sigs.k8s.io/kustomize/api/types"
+
+	stacksv1alpha1 "github.com/crossplaneio/resourcepacks/api/v1alpha1"
+	"github.com/crossplaneio/resourcepacks/pkg/controllers"
+	"github.com/crossplaneio/resourcepacks/pkg/operations/kustomize"
+)
+
+const (
+	KustomizeEngine = "kustomize"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func main() {
+	var (
+		// top level app definition
+		app = kingpin.New(filepath.Base(os.Args[0]), "An open source multicloud control plane.").DefaultEnvars()
+		//debug      = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
+
+		// The default controller mode.
+		controllerCmd = app.Command(filepath.Base(os.Args[0]), "An open source multicloud control plane.").Default()
+
+		// Configuration for the reconciler.
+		reconcileCmd                = controllerCmd.Command("reconcile", "Reconcile a Custom Resource")
```

Comment:
Is this going to be running in a persistent process? `reconcile` feels like a one-time reconciliation to me, probably because of the naming convention that the controller-runtime reconcilers use for their `Reconcile` method.

Reply:
> Is this going to be running in a persistent process?

Yes.

Actually I wrote that part before we decided that there will be one CRD in `TemplateStack` CR so, `reconcile` subcommand would take GVK as flag. I think we can just get rid of it. The call would look like:
```
$ /manager --template-stack-name some-name --template-stack-namespace default
```